### PR TITLE
Sensor details page

### PR DESCRIPTION
**Pull request by artem713**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/8" target="_blank">ReviewNinja

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "server": "node --harmony_destructuring --harmony_rest_parameters server/index.js",
     "client": "webpack-dev-server --inline --watch",
     "test": "karma start",
+    "tdd": "karma start --no-single-run --auto-watch --reporters nyan",
```
Comment (kosich) [inline package.json:11]: usually its better to add such improvements as a separate PR: it makes it easier to manage git.
at least they can be faster merged to common branch

File: src/app/sh-request-options.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import {BaseRequestOptions, RequestOptionsArgs} from 'angular2/http';
+
+const DEFAULT_URL = 'http://localhost';
+const DEFAULT_PORT = '3000';
```
Comment (kosich) [inline src/app/sh-request-options.js]: !fix should be moved out to a config file

File: server/index.js [hunk 3]
```diff
@@ -22,6 +22,22 @@ app.post('/emulate', (req, res) => {
     res.redirect('/');
 });
 
+app.get('/sensor/api/sensor/get/:id', (req, res) => {
```
Comment (kosich) [inline server/index.js]: why do we have such strange prefix `/sensor/api/sensor/`?

File: src/app/sensor-detail/sensor-detail.service.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    it('should get sensor data from the server', () => {
+        const idMock = 'mock';
+        sut.get(idMock);
+        expect(httpMock.get).toHaveBeenCalledWith(`/sensors/${idMock}`);
+    });
+
+    it('should save sensor', () => {
+        const sensorMock = {_id: 'mock'};
+        const body = JSON.stringify(sensorMock);
```
Comment (kosich) [inline src/app/sensor-detail/sensor-detail.service.test.js]: !fix stringifying is a responsibility of the http service

File: src/app/sh-http.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import {Injectable} from 'angular2/core';
```
Comment (kosich) [inline src/app/sh-http.js]: !fix services should follow naming conventions:
file should have `.service` and class should have `Service` suffixes

Comment (kosich) [inline src/app/sh-http.js]: !fix move services for BE communication to a separate folder

File: package.json (hunk 1 above)
Comment (artem713) [inline package.json:11]: I'll take it as a note in my next commits, thanx)

File: server/index.js (hunk 3 above)
Comment (artem713) [inline server/index.js]: Actually we don't need it any more, I'll remove it

File: src/app/sensor-detail/sensor-detail.service.test.js (hunk 4 above)
Comment (artem713) [inline src/app/sensor-detail/sensor-detail.service.test.js]: Well, yes, but this constant is just for checking function parameters when it called. I'm not sending body as a parameter for `save` method.

Comment (kosich) [inline src/app/sensor-detail/sensor-detail.service.test.js]: oops, missed
was actually addressing line 16 in sensor-detail.service.js
`const body = JSON.stringify(sensor);`

Comment (kosich) [inline src/app/sensor-detail/sensor-detail.service.test.js]: !fixed

File: src/app/sh-http.js (hunk 5 above)
Comment (kosich) [inline src/app/sh-http.js]: !fixed

File: src/app/sh-request-options.js (hunk 2 above)
Comment (kosich) [inline src/app/sh-request-options.js]: !fixed

**Pull request merged**